### DISCUSSION TITLE
DataGrid - CellTemplateSelector is missing

### DISCUSSION THREAD

**Discussion by Symbai** [AvaloniaUI/Avalonia#15381] [Q&A]
WPF's DataGrid CellTemplateSelector property is missing in Avalonia. Is there any existing equivalent? It is needed to show different template based on for example what item the column is bound to.

**Comment by IngvarX** [comment]
There is a hack that allows to use different data templates for different types in cell. You should override `DataGridTemplateColumn.CellTemplate` with `ContentControl` and define your templates inside `<ContentControl.DataTemplates>`

**Comment by BuriKizilkaya** [comment]
Is there any progress without the hack?

**Comment by rabbitism** [comment]
> Is there any progress without the hack?

Avalonia IDataTemplate naturally act as both DataTemplate and DataTemplateSelector.

**Comment by timunie** [comment]
I think this is the answer here as well: https://github.com/AvaloniaUI/Avalonia/discussions/15380#discussioncomment-9116096

In my opinion, this is by design in Avalonia and should be a Q&A instead.

**Comment by BuriKizilkaya** [comment]
In the Datagrid we can add a DataGridTemplateColumn, which accepst one DataTemplate (see [here](https://docs.avaloniaui.net/docs/reference/controls/detailed-reference/datagrid/data-grid-template-columns)). 

If I want to support multiple DataTemplates the trick is to add a ContentControl inside the CellTemplate and so via the ViewLocator or local DataTemplates I can load different views. What I have is that the DataGridTemplateColumn directly accepts multiple DataTemplates and a binding to the property that can be different ViewModels.

**Reply by timunie** [reply]
Please see: https://github.com/AvaloniaUI/Avalonia/discussions/15380#discussioncomment-9116096
